### PR TITLE
Bugfix/soil absorption accounting

### PR DESCRIPTION
**Pull request by laestrada**
### Name and Institution (Required)

Name: Lucas Estrada
Institution: Harvard ACMG

### Describe the update

This bugfix addresses #210. Effectively, the soil absorption sink is not optimized in the inversion and we were not properly accounting for this when calculating the posterior emissions.

### CODE REVIEW

**Review by msulprizio — APPROVED** [review]
Thanks for these fixes. Looks good to me!

**Review by megan-he — APPROVED** [review]
Looks good, thanks Lucas!

File: src/components/kalman_component/prepare_sf.py [hunk 1]
```diff
@@ -81,9 +82,11 @@ def prepare_sf(config_path, period_number, base_directory, nudge_factor):
...
 
             # Get posterior emissions multiplied up to current period p, and apply nudging
-            current_posterior_emis = (
-                posterior_p["ScaleFactor"] * sf["ScaleFactor"] * original_emis
-            )
+            # Account for non-optimized soil sink by calling get_posterior_emissions()
+            posterior_scale_ds["ScaleFactor"] = posterior_p["ScaleFactor"] * sf["ScaleFactor"]
+            posterior_emis_ds = get_posterior_emissions(original_emis_ds, posterior_scale_ds)
+            current_posterior_emis = posterior_emis_ds["EmisCH4_Total"].isel(time=0, drop=True)
+
```
Comment (laestrada) [inline src/components/kalman_component/prepare_sf.py:89]: One point I am unsure of is whether we should be removing the soil sink in the calculation of the nudged scale factor below (line 108). I think the answer is yes, but would like to check what @megan-he, @msulprizio, and especially @djvaron think.

Comment (djvaron) [reply src/components/kalman_component/prepare_sf.py:89]: This is a good question! I _think_ what we want to do is:

1. Remove soil sink from current posterior and original prior
2. Do weighted average of current posterior and original prior
3. Apply lambda scaling
4. Compute scale factors to apply in next period (i.e., sf["ScaleFactor"] on line 108)

File: src/components/kalman_component/prepare_sf.py [hunk 2]
```diff
@@ -5,7 +5,19 @@
...
+    """
+    Remove soil absorption from total emissions and return the new total.
+    
+    Arguments
+        emis [xr.Dataset] : HEMCO emissions dataset
+    Returns
+        [xr.DataArray] : Total emission from all sources except soil absorption
+    """
+    ds = emis.copy()
+    ds["EmisCH4_Total"] = ds["EmisCH4_Total"] - ds["EmisCH4_Soil"]
```
Comment (djvaron) [inline src/components/kalman_component/prepare_sf.py:18]: Shouldn't it be `EmisCH4_SoilAbsorb` ?

**Pull request merged**